src/Microsoft.PowerShell.ConsoleHost/WindowsTaskbarJumpList/TaskbarJumpList.cs

```diff
@@ -4,14 +4,55 @@
 using System;
 using System.Diagnostics;
 using System.Reflection;
+using System.Threading;
 
 using static Microsoft.PowerShell.ComInterfaces;
 
 namespace Microsoft.PowerShell
 {
     internal static class TaskbarJumpList
     {
-        internal static void CreateElevatedEntry(string title)
+        // Creating a JumpList entry takes around 55ms when the PowerShell process is interactive and
+        // owns the current window (otherwise it does a fast exit anyway). Since there is no 'GET' like API,
+        // we always have to execute this call because we do not know if it has been created yet.
+        // The JumpList does persist as long as the filepath of the executable does not change but there
+        // could be disruptions to it like e.g. the bi-annual Windows update, we decided to
+        // not over-optimize this and always create the JumpList as a non-blocking background task instead.
+        internal static void CreateRunAsAdministratorJumpList()
+        {
+            try
+            {
+                // Some COM APIs are STA only, therefore the executing thread must run in STA
+                if (Thread.CurrentThread.GetApartmentState() == ApartmentState.STA)
+                {
+                    TaskbarJumpList.CreateElevatedEntry(ConsoleHostStrings.RunAsAdministrator);
+                }
+                else
+                {
+                    var thread = new Thread(() =>
+                    {
+                        try
+                        {
+                            TaskbarJumpList.CreateElevatedEntry(ConsoleHostStrings.RunAsAdministrator);
+                        }
+                        catch
+                        {
+                            // Due to COM threading complexity there might be sporadic failures but they can
+                            // be ignored as creating the JumpList is not critical and persists after its first creation.
+                        }
+                    });
+                    thread.SetApartmentState(ApartmentState.STA);
+                    thread.Start();
+                }
+            }
+            catch
+            {
+                // Due to COM threading complexity there might be sporadic failures but they can
+                // be ignored as creating the JumpList is not critical and persists after its first creation.
+            }
```

Comment:
The outer try/catch doesn't seem to be necessary, since it only tries to start a new thread.

Reply:
We could still encounter a `OutOfMemoryException`
https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread.start?view=netcore-3.0
Yes, the system would be in a bad system at this point in time but because the JumpList is not essential to the functionality of PowerShell, I'd still let it proceed. What do you think, should we just catch the `OutOfMemoryException` or not catch at all?

Reply:
You can always encounter non-resumable exceptions with the process being in a bad state, as you discovered yourself with this very bug. The point is, are you *expecting* OutOfMemory?

There are two kinds of OutOfMemory, first those when you do a large allocation which fails. Since the allocation failed you can catch and resume because there is still plenty of memory free. If you expect this kind of OOM then you should catch it explicitely. The other kind of OOM is when you are *actually* OOM; catching it will not help you because the next allocation will fail again. The system always does a full GC before throwing OOM so you don't save anything by guarding against the second kind, the only thing you can do at this point is clean up in a finalizer.

The jumplist logic probably very memory-heavy. Creating a thread is kinda expensive, but I'd argue that if you have that little memory left then you can't do anything meaningfull in the process anymore either.

Reply:
The outer try/catch is not even protecting the jumplist logic, it's protecting the `thread.Start` call. If we got any exception when doing that, then it's likely fatal anyways.

Reply:
OK, I removed the outer catch block of the thread creation.